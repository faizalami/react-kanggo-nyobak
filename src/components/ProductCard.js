/** @jsxImportSource @emotion/react */
import styled from '@emotion/styled';
import { margin, padding, rounded, width } from './utilities';
import { Flex } from './FlexGrid';
import { gray } from './variables';
import { CircleButton, CircleButtonLink } from './Buttons';
import { ReactComponent as EyeIcon } from '../icons/eye.svg';
import { ReactComponent as PencilIcon } from '../icons/pencil.svg';
import { ReactComponent as TrashIcon } from '../icons/trash.svg';

const CardWrapper = styled(Flex)`
  border: 1px solid ${gray};
  ${rounded}
`;

const CardThumbnail = styled.img`
  aspect-ratio: 1/1;
  object-position: center;
  object-fit: cover;
  ${rounded}
  ${width.full}
`;

export default function ProductCard ({ id, name, picture, price }) {
  return (
    <CardWrapper column>
      <CardThumbnail src={picture} alt={name} loading="lazy" width={300} height={300}/>

      <Flex justifyContent="center" css={[padding.y2, width.full]}>
        <CircleButtonLink to={`/product/${id}`}>
          <EyeIcon/>
        </CircleButtonLink>
        <CircleButtonLink to={`/product/edit/${id}`} css={margin.x2}>
          <PencilIcon/>
        </CircleButtonLink>
        <CircleButton variant="danger">
          <TrashIcon/>
        </CircleButton>
      </Flex>

      <h3 css={[margin.b0, margin.x2]}>{name}</h3>
      <p css={[margin.b2, margin.x2]}>Rp {price}</p>
    </CardWrapper>
  );
}
