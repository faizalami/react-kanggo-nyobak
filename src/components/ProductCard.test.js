import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ProductCard from './ProductCard';

test('ProductCard match with snapshot', () => {
  const dummyProduct = {
    id: 1,
    name: 'Test',
    picture: null,
    price: 20000,
  };

  expect(render(
    <MemoryRouter>
      <ProductCard {...dummyProduct} />
    </MemoryRouter>
  )).toMatchSnapshot();
})
