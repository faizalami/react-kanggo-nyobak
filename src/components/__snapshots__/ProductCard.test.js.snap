// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border: 1px solid #d1d5db;
  border-radius: 0.5rem;
}

.emotion-1 {
  aspect-ratio: 1/1;
  object-position: center;
  object-fit: cover;
  border-radius: 0.5rem;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  width: 100%;
}

.emotion-3 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: white;
  background-color: #4f46e5;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  width: 2.5rem;
  height: 2.5rem;
  padding: 0rem;
}

.emotion-3:hover,
.emotion-3:focus {
  background-color: #4338ca;
}

.emotion-3 svg {
  width: 1.5rem;
  height: 1.5rem;
}

.emotion-4 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: white;
  background-color: #4f46e5;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  width: 2.5rem;
  height: 2.5rem;
  padding: 0rem;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
}

.emotion-4:hover,
.emotion-4:focus {
  background-color: #4338ca;
}

.emotion-4 svg {
  width: 1.5rem;
  height: 1.5rem;
}

.emotion-5 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: white;
  background-color: #dc2626;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  width: 2.5rem;
  height: 2.5rem;
  padding: 0rem;
}

.emotion-5:hover,
.emotion-5:focus {
  background-color: #b91c1c;
}

.emotion-5 svg {
  width: 1.5rem;
  height: 1.5rem;
}

.emotion-6 {
  margin-bottom: 0rem;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
}

.emotion-7 {
  margin-bottom: 0.5rem;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
}

<body>
    <div>
      <div
        class="emotion-0"
        data-testid="product-card"
      >
        <img
          alt="Test"
          class="emotion-1"
          height="300"
          loading="lazy"
          width="300"
        />
        <div
          class="emotion-2"
        >
          <a
            aria-label="Detail"
            class="emotion-3"
            href="/product/1"
          >
            <svg>
              eye.svg
            </svg>
          </a>
          <a
            aria-label="Edit"
            class="emotion-4"
            href="/product/edit/1"
          >
            <svg>
              pencil.svg
            </svg>
          </a>
          <button
            aria-label="Delete"
            class="emotion-5"
          >
            <svg>
              trash.svg
            </svg>
          </button>
        </div>
        <h3
          class="emotion-6"
        >
          Test
        </h3>
        <p
          class="emotion-7"
        >
          Rp 
          20000
        </p>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border: 1px solid #d1d5db;
  border-radius: 0.5rem;
}

.emotion-1 {
  aspect-ratio: 1/1;
  object-position: center;
  object-fit: cover;
  border-radius: 0.5rem;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  width: 100%;
}

.emotion-3 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: white;
  background-color: #4f46e5;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  width: 2.5rem;
  height: 2.5rem;
  padding: 0rem;
}

.emotion-3:hover,
.emotion-3:focus {
  background-color: #4338ca;
}

.emotion-3 svg {
  width: 1.5rem;
  height: 1.5rem;
}

.emotion-4 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: white;
  background-color: #4f46e5;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  width: 2.5rem;
  height: 2.5rem;
  padding: 0rem;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
}

.emotion-4:hover,
.emotion-4:focus {
  background-color: #4338ca;
}

.emotion-4 svg {
  width: 1.5rem;
  height: 1.5rem;
}

.emotion-5 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: white;
  background-color: #dc2626;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  width: 2.5rem;
  height: 2.5rem;
  padding: 0rem;
}

.emotion-5:hover,
.emotion-5:focus {
  background-color: #b91c1c;
}

.emotion-5 svg {
  width: 1.5rem;
  height: 1.5rem;
}

.emotion-6 {
  margin-bottom: 0rem;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
}

.emotion-7 {
  margin-bottom: 0.5rem;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
}

<div>
    <div
      class="emotion-0"
      data-testid="product-card"
    >
      <img
        alt="Test"
        class="emotion-1"
        height="300"
        loading="lazy"
        width="300"
      />
      <div
        class="emotion-2"
      >
        <a
          aria-label="Detail"
          class="emotion-3"
          href="/product/1"
        >
          <svg>
            eye.svg
          </svg>
        </a>
        <a
          aria-label="Edit"
          class="emotion-4"
          href="/product/edit/1"
        >
          <svg>
            pencil.svg
          </svg>
        </a>
        <button
          aria-label="Delete"
          class="emotion-5"
        >
          <svg>
            trash.svg
          </svg>
        </button>
      </div>
      <h3
        class="emotion-6"
      >
        Test
      </h3>
      <p
        class="emotion-7"
      >
        Rp 
        20000
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
